utils/src/main/java/uk/gov/pay/commons/utils/metrics/DatabaseMetricsService.java

```diff
@@ -0,0 +1,136 @@
+package uk.gov.pay.commons.utils.metrics;
+
+import com.codahale.metrics.Counter;
+import com.codahale.metrics.Gauge;
+import com.codahale.metrics.MetricRegistry;
+import uk.gov.pay.commons.utils.db.JDBCConnectionProvider;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.Set;
+
+import static java.lang.String.format;
+
+public class DatabaseMetricsService {
+
+    private final Set<PostgresMetric> metrics = Set.of(
+            new PostgresGaugeMetric("numbackends", 0L),
+            new PostgresCounterMetric("xact_commit", 0L),
+            new PostgresCounterMetric("xact_rollback", 0L),
+            new PostgresCounterMetric("blks_read", 0L),
+            new PostgresCounterMetric("blks_hit", 0L),
+            new PostgresCounterMetric("tup_returned", 0L),
+            new PostgresCounterMetric("tup_fetched", 0L),
+            new PostgresCounterMetric("tup_inserted", 0L),
+            new PostgresCounterMetric("tup_updated", 0L),
+            new PostgresCounterMetric("tup_deleted", 0L),
+            new PostgresCounterMetric("conflicts", 0L),
+            new PostgresCounterMetric("temp_files", 0L),
+            new PostgresCounterMetric("temp_bytes", 0L),
+            new PostgresCounterMetric("deadlocks", 0L),
+            new PostgresCounterMetric("blk_read_time_ns", 0L),
+            new PostgresCounterMetric("blk_write_time_ns", 0L));
+
+    private final String databaseName;
+    private final JDBCConnectionProvider jdbcConnectionProvider;
+    private Integer statsHealthy = 0;
+
+    public DatabaseMetricsService(JDBCConnectionProvider jdbcConnectionProvider, MetricRegistry metricRegistry, String databaseName) {
```

Comment:
I think maybe we should have a second constructor which takes a `DataSourceFactory` so that the user of this class doesn't need to import or know about `JDBCConnectionProvider` which is only there to facilitate the tests